modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -25,7 +26,7 @@ package object engine {
     * This represents an actual real-world action to be done in the underlying
     * systems.
     */
-  type Action = Task[Result]
+  type Action = Kleisli[Task, ActionMetadata, Result]
```

Comment:
Why not `case class Action(metadata: ActionMetadata, task: Task[Result]`)? 

I like the `Kleisli` because it's a nice pattern to avoid passing read-only parameters all over the place but I don't see in this PR any code reading the metadata using the `Kleisli` so where are you expecting to use it?

In practice the `Kleisli` represents the Reader monad transformer and I got the impression we wanted to avoid them. I'm fine with monad transformers, so does this mean we can start using them without constraints?